parquet-thrift/src/main/java/parquet/thrift/BufferedProtocolReadToWrite.java

```diff
@@ -394,10 +404,42 @@ public String toDebugString() {
           return "f=" + currentField.id + "<t=" + typeName(currentField.type) + ">: ";
         }
       });
-      hasFieldsIgnored |= readOneValue(in, field.type, buffer, expectedField.getType());
+      boolean wasIgnored = readOneValue(in, field.type, buffer, expectedField.getType());
+      if (!wasIgnored) {
+        childFieldsPresent++;
+      }
+      hasFieldsIgnored |= wasIgnored;
       in.readFieldEnd();
       buffer.add(FIELD_END);
     }
+
+    // check that union had exactly 1 (no more no less) child fields.
```

Comment:
This method is becoming increasingly complicated, can we move the code below to a private method?